packages/pages/src/dev/server/ssr/getLocalData.ts

```diff
@@ -42,30 +62,44 @@ export const getLocalDataManifest = async (): Promise<LocalDataManifest> => {
     );
 
     const featureName = data.__?.name?.toString();
-    const entityId = data.id?.toString();
-
     // Check for a feature name because the localData folder will have a mapping.json
     // file which should not be included in the manifest.
-    if (featureName) {
-      if (entityId) {
-        localDataManifest.entity.set(featureName, [
-          ...(localDataManifest.entity.get(featureName) || []),
-          entityId,
-        ]);
-      } else {
-        // The lack of an entityId signifies that this is a static template.
-        localDataManifest.static.push(featureName);
+    if (!featureName) {
+      continue;
+    }
+    const entityId = data.id?.toString();
+    const slug = data.slug?.toString();
+    if (entityId) {
+      localDataManifest.entity.set(featureName, [
+        ...(localDataManifest.entity.get(featureName) || []),
+        { entityId, slug },
+      ]);
+    } else {
+      // The lack of an entityId signifies that this is a static template.
+      const templateModuleInternal = await findTemplateModuleInternal(
```

Comment:
Why do we need to do this now when we didn't before? It doesn't seem to be related to slugs this that's handled above.

Reply:
This is to generate the URLs for static pages. Previously, the static page URLs would be the name of the feature, however the template name is typically not the URL that would be used in prod.

For instance in the pages react starter's `robots.tsx`, the feature name is `'turtlehead-tacos'`, however the getPath returns `'index.html'`. Previously, the corresponding URL in the index page would be `'turtlehead-tacos'`, but now it is `'index.html'`

Reply:
Makes sense.